From d52d14fd1ff38e7c64813f4d83d230b60d849d2c Mon Sep 17 00:00:00 2001
From: tanghui <tanghui20@huawei.com>
Date: Wed, 16 Feb 2022 15:32:59 +0800
Subject: [PATCH 0789/2065] [Huawei] sched: Introduce dynamic affinity for cfs
 scheduler

Offering: HULK
hulk inclusion
category: feature
bugzilla: 186575

--------------------------------

Dynamic affinity set preferred cpus for task. When the utilization of
taskgroup's preferred cpu is low, task only run in cpus preferred to
enhance cpu resource locality and reduce interference between task cgroups,
otherwise task can burst preferred cpus to use external cpu within
cpus allowed.

Signed-off-by: tanghui <tanghui20@huawei.com>
---
 init/Kconfig | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/init/Kconfig b/init/Kconfig
index 895e0ef85f73..e89ed78fc816 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -995,6 +995,16 @@ config RT_GROUP_SCHED
 
 endif #CGROUP_SCHED
 
+config QOS_SCHED_DYNAMIC_AFFINITY
+	bool "qos dynamic affinity"
+	depends on CPUSETS
+	default n
+	help
+	 This feature lets you allocate preferred cpus to taskgroup. If enabled,
+	 it will make taskgroup only to use preferred cpus when cpu utilization
+	 of taskgroup is below threshold setted, otherwise make taskgroup to use
+	 cpus allowed.
+
 config UCLAMP_TASK_GROUP
 	bool "Utilization clamping per group of tasks"
 	depends on CGROUP_SCHED
-- 
2.17.1

